storage/watchable_store_test.go

```diff
@@ -222,3 +239,109 @@ func TestUnsafeAddWatching(t *testing.T) {
 		}
 	}
 }
+
+func TestNewMapWatchingToEventMap(t *testing.T) {
+	k0, k1, k2 := []byte("foo0"), []byte("foo1"), []byte("foo2")
+	v0, v1, v2 := []byte("bar0"), []byte("bar1"), []byte("bar2")
+
+	ws := []*watching{
+		{key: k0},
+		{key: k1},
+		{key: k2},
+	}
+
+	evs := []storagepb.Event{
+		{
+			Type: storagepb.PUT,
+			Kv:   &storagepb.KeyValue{Key: k0, Value: v0},
+		},
+		{
+			Type: storagepb.PUT,
+			Kv:   &storagepb.KeyValue{Key: k1, Value: v1},
+		},
+		{
+			Type: storagepb.PUT,
+			Kv:   &storagepb.KeyValue{Key: k2, Value: v2},
+		},
+	}
+
+	tests := []struct {
+		sync map[string]map[*watching]struct{}
+		evs  []storagepb.Event
+
+		wwe map[*watching][]storagepb.Event
+	}{
+		// no watching in sync, some events should return empty wwe
+		{
+			map[string]map[*watching]struct{}{},
+			evs,
+			map[*watching][]storagepb.Event{},
+		},
+
+		// one watching in sync, one event that does not match the key of that
+		// watching should return empty wwe
+		{
+			map[string]map[*watching]struct{}{
+				string(k2): {ws[2]: struct{}{}},
+			},
+			evs[:1],
+			map[*watching][]storagepb.Event{},
+		},
+
+		// one watching in sync, one event that matches the key of that
+		// watching should return wwe with that matching watching
+		{
+			map[string]map[*watching]struct{}{
+				string(k1): {ws[1]: struct{}{}},
+			},
+			evs[1:2],
+			map[*watching][]storagepb.Event{
+				ws[1]: evs[1:2],
+			},
+		},
+
+		// two watchings in sync that watches two different keys, one event
+		// that matches the key of only one of the watching should return wwe
+		// with the matching watching
+		{
+			map[string]map[*watching]struct{}{
+				string(k0): {ws[0]: struct{}{}},
+				string(k2): {ws[2]: struct{}{}},
+			},
+			evs[2:],
+			map[*watching][]storagepb.Event{
+				ws[2]: evs[2:],
+			},
+		},
+
+		// two watchings in sync that watches the same key, two events that
+		// match the keys should return wwe with those two watchings
+		{
+			map[string]map[*watching]struct{}{
+				string(k0): {ws[0]: struct{}{}},
+				string(k1): {ws[1]: struct{}{}},
+			},
+			evs[:2],
+			map[*watching][]storagepb.Event{
+				ws[0]: evs[:1],
+				ws[1]: evs[1:2],
+			},
+		},
+	}
+
```

Comment:
@xiang90 Just changed the variable names shorter, as you suggested. Please take a look again.